### DISCUSSION TITLE
IDE setup

### DISCUSSION THREAD

**Discussion by ArneKoeckeritz** [maplibre/maplibre-native#325] [Q&A]
Hi there,

what's a recommended IDE set up to start learning the code base including compiling and building?
I'm an Android developer who wants to track down and report some bugs I see in Crashlytics.

Best
Arne

**Comment by wipfli** [comment]
I don't know but if you find out, please share it. I am happy to review any docs pull request...

**Comment by wipfli** [comment]
You can build the library and run some tests inside a docker container. This container is used by CI. Here are a few steps for android:

```bash
docker run --rm -it ghcr.io/maplibre/android-ndk-r21b
git clone --recurse-submodules --depth 1 https://github.com/maplibre/maplibre-gl-native.git
cd maplibre-gl-native/platform/android
make android-check
```

This is probably not the best workflow, but it is a starting point....
